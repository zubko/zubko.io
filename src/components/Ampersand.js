/**
 * Component which renders a beautiful & symbol
 * @flow
 */

import React from 'react';
import { css, Global } from '@emotion/react';
import styled from '@emotion/styled';

const globalCss = css`
  @font-face {
    font-family: Ampersand;
    src: local('Baskerville-Italic'), local('GoudyOldStyleT-Italic'),
      local('Palatino-Italic'), local('BookAntiqua-Italic');
    unicode-range: U+26;
  }
`;

const TextWithAmp = styled.span`
  font-family: Ampersand, sans-serif;
  font-size: 115%;
`;

export default () => (
  <>
    <Global styles={globalCss} />
    <TextWithAmp>&</TextWithAmp>
  </>
);
