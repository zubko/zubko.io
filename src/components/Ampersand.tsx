/**
 * Component which renders a beautiful "&" symbol
 */

import { css, Global } from '@emotion/react';
import styled from '@emotion/styled';
import React from 'react';

const globalCss = css`
  @font-face {
    font-family: Ampersand;
    src: local('Baskerville-Italic'), local('GoudyOldStyleT-Italic'),
      local('Palatino-Italic'), local('BookAntiqua-Italic');
    unicode-range: U+26;
  }
`;

const TextWithAmp = styled.span`
  font-family: Ampersand, sans-serif;
  font-size: 115%;
`;

const Ampersand = () => (
  <>
    <Global styles={globalCss} />
    <TextWithAmp>&</TextWithAmp>
  </>
);

export default Ampersand;
