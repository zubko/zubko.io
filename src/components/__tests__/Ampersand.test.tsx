import React from "react";
import renderer from "react-test-renderer";

import Ampersand from "../Ampersand";

describe("Ampersand", () => {
  it("renders without regressions", () => {
    const tree = renderer.create(<Ampersand />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
