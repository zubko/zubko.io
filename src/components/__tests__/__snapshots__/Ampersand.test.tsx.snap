// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ampersand renders without regressions 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@font-face{font-family:Ampersand;src:local(\\"Baskerville-Italic\\"),local(\\"GoudyOldStyleT-Italic\\"),local(\\"Palatino-Italic\\"),local(\\"BookAntiqua-Italic\\");unicode-range:U+26;}",
      }
    }
    data-emotion="css-global 1jhw1vb"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1an0k18{font-family:Ampersand,sans-serif;font-size:115%;}",
      }
    }
    data-emotion="css 1an0k18"
  />,
  <span
    className="css-1an0k18"
  >
    &
  </span>,
]
`;
